app/views/mod.py

```diff
@@ -0,0 +1,225 @@
+from flask_login import login_required, current_user
+from flask_babel import _
+from .. import misc
+from ..forms import TOTPForm, LogOutForm, UseInviteCodeForm, AssignUserBadgeForm, EditModForm, BanDomainForm
+from ..models import UserMetadata, User, Sub, SubPost, SubPostComment, SiteMetadata
+from ..models import User, Sub, SubMod, SubPost, SubPostComment, UserMetadata, SubPostReport, SubPostCommentReport
+from ..misc import engine, getSubReports, getModSubs, engine
+
+bp = Blueprint('mod', __name__)
+
+
+@bp.route('/mod/auth', methods=['GET', 'POST'])
+@login_required
+def auth():
+    if not current_user.is_mod:
+        abort(404)
+    form = TOTPForm()
+    try:
+        user_secret = UserMetadata.get((UserMetadata.uid == current_user.uid) & (UserMetadata.key == 'totp_secret'))
+    except UserMetadata.DoesNotExist:
+        return engine.get_template('mod/totp.html').render({'authform': form, 'error': _('No TOTP secret found.')})
+    if form.validate_on_submit():
+        totp = TOTP(user_secret.value)
+        if totp.verify(form.totp.data):
+            session['apriv'] = time.time()
+            return redirect(url_for('mod.index'))
+        else:
+            return engine.get_template('mod/totp.html').render(
+                {'authform': form, 'error': _('Invalid or expired token.')})
+    return engine.get_template('mod/totp.html').render({'authform': form, 'error': None})
+
+
+@bp.route('/logout', methods=['POST'])
+@login_required
+def logout():
+    if not current_user.is_mod:
+        abort(404)
+    form = LogOutForm()
+    if form.validate():
+        del session['apriv']
+    return redirect(url_for('mod.index'))
+
+
+@bp.route("/")
+@login_required
+def index():
+    """ WIP: Mod Dashboard """
+
+    if not current_user.is_mod:
+        abort(404)
+
+    subs = getModSubs(current_user.uid)
+
+    updated_subs = []
+
+    for sub in subs:
+        # get the sub sid
+        this_sub = Sub.select().where(Sub.sid == sub.sid).get()
+        sid = str(this_sub.sid)
+        # use sid to getSubReports
+        reports = getSubReports(sid)
+        # count open and closed reports
+        open_reports_count = reports['open'].count()
+        closed_reports_count = reports['closed'].count()
+        # add open_report_count and closed_report_count as properties on sub
+
+        updated_sub = {
+            'name': str(this_sub.name) ,
+            'subscribers': str(this_sub.subscribers),
+            'open_reports_count': open_reports_count,
+            'closed_reports_count': closed_reports_count
+        }
+
+        updated_subs = updated_subs + [updated_sub]
+    return render_template('mod/mod.html', subs=updated_subs)
+
+
+@bp.route("/reports", defaults={'page': 1})
+@bp.route("/reports/<int:page>")
+@login_required
+def reports(page):
+    """ WIP: Open Report Queue """
+
+    if not current_user.is_mod:
+        abort(404)
+
+    mod_subs = getModSubs(current_user.uid)
+
+    def isSubMod(sid, mod_subs):
```

Comment:
This function seems to always return `True`, effectively showing all reports to everybody. It'd probably be more efficient anyway to filter by joining to the SubMod table and then just using a where

Reply:
lol yeah that's much better... I think I just never finished doing whatever I was doing there! :joy: